src/App/Pages/Accounts/TwoFactorPageViewModel.cs

```diff
@@ -208,6 +224,51 @@ public void Load()
             ShowContinue = !(SelectedProviderType == null || DuoMethod || Fido2Method);
         }
 
+        private async Task DuoFramelessAuthenticateAsync()
+        {
+            await _deviceActionService.ShowLoadingAsync(AppResources.Validating);
+            var providerData = _authService.TwoFactorProvidersData[SelectedProviderType.Value];
+            var url = providerData["AuthUrl"] as string;
+            WebAuthenticatorResult authResult;
+            try
+            {
+                authResult = await WebAuthenticator.AuthenticateAsync(new WebAuthenticatorOptions
+                {
+                    Url = new Uri(url),
+                    CallbackUrl = new Uri("bitwarden://duo-callback")
+                });
+            }
+            catch (TaskCanceledException)
+            {
+                // user canceled
+                await _deviceActionService.HideLoadingAsync();
+                return;
+            }
+
+            string response = null;
+            if (authResult != null && authResult.Properties.TryGetValue("code", out var resultData))
+            {
+                response = Uri.UnescapeDataString(resultData);
+            }
+            if (!string.IsNullOrWhiteSpace(response))
+            {
+                Token = response;
+                await SubmitAsync(false);
```

Comment:
🤔 I'm a bit concerned here on how the interaction is when the loading is being presented and then after it goes to `SubmitAsync` the user suddenly doesn't have internet connection which internally it will try to [show a dialog](https://github.com/bitwarden/mobile/blob/6eb7941d6ca4540f5be904095b9cee538cc10813/src/App/Pages/Accounts/TwoFactorPageViewModel.cs#L347) and perhaps it could be a problem with both of them trying to be shown at the same time. Could you test that? I know this may be also an issue with all the other providers as well so maybe the problem is bigger than just here.
If so, we should hide the loading before attempting to show any dialogs inside the `SubmitAsync(...)` method

Reply:
Yup it's a problem because when it shows the internet error dialog it never dismisses the old loading dialog. Change the code to first hide the loading and then call the `SubmitAsync(true)`